plugins/lookup/hashi_vault.py

```diff
@@ -1,109 +1,229 @@
+# (c) 2020, Brian Scholer (@briantist)
 # (c) 2015, Jonathan Davila <jonathan(at)davila.io>
 # (c) 2017 Ansible Project
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import (absolute_import, division, print_function)
 __metaclass__ = type
 
-DOCUMENTATION = '''
+DOCUMENTATION = """
   lookup: hashi_vault
-  author: Jonathan Davila <jdavila(at)ansible.com>
-  short_description: retrieve secrets from HashiCorp's vault
+  author:
+    - Jonathan Davila <jdavila(at)ansible.com>
+    - Brian Scholer (@briantist)
+  short_description: Retrieve secrets from HashiCorp's vault
   requirements:
     - hvac (python library)
+    - hvac 0.7.0+ (for namespace support)
+    - hvac 0.9.6+ (to avoid all deprecation warnings)
+    - botocore (only if inferring aws params from boto)
+    - boto3 (only if using a boto profile)
   description:
-    - retrieve secrets from HashiCorp's vault
+    - Retrieve secrets from HashiCorp's vault.
   notes:
     - Due to a current limitation in the HVAC library there won't necessarily be an error if a bad endpoint is specified.
-    - As of Ansible 2.10, only the latest secret is returned when specifying a KV v2 path.
+    - As of Ansible 2.10, only the latest version of a secret is returned when specifying a KV v2 path.
```

Comment:
@gundalow how should we write things like this? `As of community.general ...`? I'm not really sure...

Reply:
I saw some discussion about this a number of weeks ago (maybe on IRC or in one of the issues tracking these things). There was no consensus at that time, but one of the suggestions was that using Ansible 2.10 was likely ok as a marker of the first collection release.

I think it also has not been decided what the initial collection version numbers will be either.

If anything is decided or the guidance changes I'm happy to update this quickly.

Reply:
I guess we can keep it for now. At least `Ansible 2.10` can be searched+replaced if we decide to change it.